scripts/firmware.py

```diff
@@ -0,0 +1,2097 @@
+        """
+        # Calculate the width dynamically based on the longest line
+        lines = logo.strip().split("\n")
+        max_width = max(len(line) for line in lines)
+        border = "=" * max_width
+
+        # Print the header with borders
+        print(border)
+        for line in lines:
+            print(Utils.colored_text(line.center(max_width), Color.GREEN))
+        print(border)
+        title = "CARTOGRAPHER FIRMWARE FLASHER"
+        version = f" v{FLASHER_VERSION}"
+        combined_title = Utils.colored_text(title, Color.CYAN) + Utils.colored_text(
+            version, Color.RED
+        )
+        print(combined_title.center(105))
+
+        # Display modes, centered
+        Utils.display_modes(args)
+
+        # Print the bottom border
+        print("=" * max_width)
+
+    @staticmethod
+    def colored_text(text: str, color: Color) -> str:
+        return f"{color.value}{text}{Color.RESET.value}"
+
+    @staticmethod
+    def error_msg(message: str) -> None:
+        print(Utils.colored_text("Error:", Color.RED), message)
+        _ = input(Utils.colored_text("\nPress Enter to continue...", Color.YELLOW))
+
+    @staticmethod
+    def success_msg(message: str) -> None:
+        print(Utils.colored_text("Success:", Color.GREEN), message)
+        _ = input(Utils.colored_text("\nPress Enter to continue...", Color.YELLOW))
+
+    @staticmethod
+    def page(title: str, width: int = PAGE_WIDTH) -> None:
+        if len(title) > width:
+            width = len(title) + 4  # Ensure width accommodates long titles with padding
+        border = "=" * width
+        print(border)
+        print(Utils.colored_text(title.center(width), Color.CYAN))
+        print(border)
+
+    @staticmethod
+    def display_modes(args: FirmwareNamespace) -> None:
+        # Map conditions to mode strings
+        mode_conditions = [
+            (args.flash, lambda: f"{(args.flash or '').upper()} MODE"),
+            (args.kseries, lambda: "K Series"),
+            (args.high_temp, lambda: "HIGH TEMP"),
+            (args.debug, lambda: "DEBUGGING"),
+            (args.branch, lambda: f"BRANCH: {(args.branch or '').upper()}"),
+            (args.type, lambda: "FLASH KATAPULT"),
+            (args.all, lambda: "ALL FIRMWARE"),
+            (is_advanced, lambda: "ADVANCED"),
+        ]
+
+        # Build modes list based on conditions
+        modes = [
+            generate_mode() for condition, generate_mode in mode_conditions if condition
+        ]
+
+        # Combine modes into a single string
+        combined_modes = " | ".join(modes)
+        Utils.show_mode(combined_modes)
+
+    @staticmethod
+    def show_mode(mode: str):
+        # Center the mode string
+        mode = mode.center(PAGE_WIDTH)
+        print(Utils.colored_text(mode, Color.RED))
+
+
+class Menu:
+    title: str
+    menu_items: Dict[int, Union["Menu.MenuItem", "Menu.Separator"]]
+
+    class MenuItem:

```

Comment:
```suggestion
    class Item:
```
I'd just call it `Item`, adding `Menu` seems redundant.

Reply:
Makes sense